packages/editor/src/store/controls.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * WordPress dependencies
+ */
+import { createRegistryControl } from '@wordpress/data';
+
+export const SELECT_PREVIOUS_BLOCK = createRegistryControl( ( registry ) => ( action ) => {
+	const { clientId } = action;
+	const { getPreviousBlockClientId } = registry.select( 'core/editor' );
+	const { selectBlock } = registry.dispatch( 'core/editor' );
+
+	selectBlock( getPreviousBlockClientId( clientId ), -1 );
+} );
+
+export const SELECT_NEXT_BLOCK = createRegistryControl( ( registry ) => ( action ) => {
```

Comment:
Can you explain why you chose to write these two actions as controls?

For me, I see controls as generic behavior (selecting, dispatching, rest api calls... but not something that looks more like an action)?

Reply:
> Can you explain why you chose to write these two actions as controls?
> 
> For me, I see controls as generic behavior (selecting, dispatching, rest api calls... but not something that looks more like an action)?

From pull request originating comment:

>I feel slightly uneasy about this implementation only because it seems to leverage controls for the mere convenient access to selectors to determine the next/previous block client IDs. I think an alternative approach could be one where the actions are handled entirely within the reducer. However, this would require that `blockSelection` have some awareness of order, which is not presently true.

Reply:
As implemented, it's very much attaching behavior to an action where a need exists to consume data from the store. I find it similar (with all the bad feelings attached) to the [Redux Thunk middleware](https://github.com/reduxjs/redux-thunk) exposing similarly the `store` as a second argument.

A problem we'll face is that this is arguably quote tempting to leverage.

I think there may be some possibility to my alternative proposal where a higher-order reducer wraps the base editor reducer, in order to share information from `editor.blocks.order` to `blockSelection`. As described by #12327, ideally these would be closer together anyways.

Reply:
Here's what I had in mind 32ea24fc95dfc425f6eaaa9637f2cd836f12c50a

Reply:
> Here's what I had in mind [32ea24f](https://github.com/WordPress/gutenberg/commit/32ea24fc95dfc425f6eaaa9637f2cd836f12c50a)

Oh, I hadn't even considered that. To be honest, it doesn't seem fundamentally different from what I've proposed as in they both leverage controls to access selectors, but as you've proposed certainly does make the implementation vastly simpler.

Reply:
The difference for me is that the "controls" should be generic things that can be shared in a reusable module across stores. (like redux middlewares or things like that)